workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -77,4 +107,143 @@ describe('WBCGroupManagemntPlugin', () => {
       );
     });
   });
+
+  describe('createIdentityPermissions', () => {
+    const sampleGroupId = 'sampleGroup';
+    const sampleGroupType: IdentityType = 'GROUP';
+
+    const sampleAction: Action = 'CREATE';
+    const sampleEffect: Effect = 'ALLOW';
+    const sampleSubjectType = 'sampleSubjectType';
+    const sampleSubjectId = 'sampleSubjectId';
+    const sampleConditions = {};
+    const sampleFields: string[] = [];
+    const sampleDescription: string = 'sampleDescription';
+
+    const mockIdentityPermission: IdentityPermission = {
+      action: sampleAction,
+      effect: sampleEffect,
+      subjectType: sampleSubjectType,
+      subjectId: sampleSubjectId,
+      identityId: sampleGroupId,
+      identityType: sampleGroupType,
+      conditions: sampleConditions,
+      fields: sampleFields,
+      description: sampleDescription
+    } as const;
+
+    const mockIdentityPermissions = [mockIdentityPermission, mockIdentityPermission];
+
+    beforeAll(() => {
+      jest.spyOn(auditService, 'write');
+    });
+
+    test('create identity permissions for valid groups', async () => {
+      mockGroupManagementPlugin.getGroupStatus = jest.fn().mockResolvedValue({ data: { status: 'active' } });
+      const mockReturnValue = {
+        data: {
+          identityPermissions: mockIdentityPermissions
+        }
+      };
+      mockDynamicAuthorizationPermissionsPlugin.createIdentityPermissions = jest
+        .fn()
+        .mockResolvedValue(mockReturnValue);
+      const params = {
+        authenticatedUser: mockUser,
+        identityPermissions: mockIdentityPermissions
+      };
+      const result = await dynamicAuthzService.createIdentityPermissions(params);
+
+      expect(result.data).toStrictEqual({
+        identityPermissions: mockIdentityPermissions
+      });
+
+      expect(auditService.write).toBeCalledWith(
+        {
```

Comment:
can we DRY up these tests and override values in `beforeEach()` methods to reduce the amount of copy and past happening in all of these tests?